clients/src/main/java/org/apache/kafka/common/MetricName.java

```diff
@@ -81,6 +81,7 @@ public MetricName(String name, String group, String description, Map<String, Str
      * @param description   A human-readable description to include in the metric
      * @param keyValue      additional key/value attributes of the metric (must come in pairs)
      */
+    @Deprecated
```

Comment:
Could we change the examples in the comment above MetricName accordingly (metricName should now be created from Metrics)?